docs/v3.x/plugins/users-permissions.md

```diff
@@ -156,53 +156,95 @@ axios
 
 Thanks to [Grant](https://github.com/simov/grant) and [Purest](https://github.com/simov/purest), you can easily use OAuth and OAuth2 providers to enable authentication in your application.
 
-Before setting up a provider, you'll need to install the `ngrok` package globally to work with providers that don't allow `localhost` redirect URIs.
+For better understanding, you may find as follows the description of the login flow. To simplify the explanation, we used `github` as the provider but it works the same for the other providers.
 
-In the following examples, the client app will be the [react login examples app](https://github.com/strapi/strapi-examples/tree/master/login-react). It will be running on `http://localhost:3000`.
+#### Understanding the login flow
+
+Let's say that strapi's backend is located at: strapi.website.fr.
+Let's say that your app frontend is located at: website.fr.
+
+1. The user goes on your frontend app (`https://website.fr`) and click on your button `connect with Github`.
+2. The frontend redirect the tab to `https://strapi.website.fr/connect/github` that calls the backend.
+3. The backend redirects the tab to the github login page where the user logs in.
+4. Once done, Github redirects the tab to `https://strapi.website.fr/connect/github/callback?code=abcdef` that calls the backend
+5. The backend uses the given `code` to get from Github an `access_token` that can be used for a period of time to make authorized requests to Github to get the user info (the email of the user of example).
+6. Then, the backend redirects the tab to the url of your choice with the param `access_token` (example: `http://website.fr/connect/github/redirect?access_token=eyfvg`)
+7. The frontend (`http://website.fr/connect/github/redirect`) calls the backend with `https://strapi.website.fr/auth/github/callback?access_token=eyfvg` that returns the strapi user profile with its `jwt`. <br> (Under the hood, the backend asks Github for the user's profile and a match is done on Github user's email address and Strapi user's email address)
+8. The frontend now possesses the user's `jwt`, with means the user is connected and the frontend can make authenticated requests to the backend!
+
+An example of a frontend app that handles this flow can be found here: [react login example app](https://github.com/strapi/strapi-examples/tree/master/login-react).
```

Comment:
Do we have to replace it with your new app?

Reply:
if https://github.com/strapi/strapi-examples/pull/164 is merged, then the url should be the same

Reply:
Okay